### PR TITLE
Add functional testing to the project

### PR DESCRIPTION
**Pull request by CohenArthur**
This PR is a collection of smaller PRs reviewed by @philberty, @flip1995 and @bjorn3. This fixes 12 out of the 13 tasks mentioned in #12, the remaining one having less impact on the project and still in need of discussing/figuring out.

### CODE REVIEW

**Comment by CohenArthur** [comment]
The tests are all being kept in a single rust test to avoid issues with environment variables and directory changes. If we can ever run tests using a single thread, without using (--test-threads=1) but by setting an option in a configuration file, I will split this test in multiple tests.

**Review by flip1995 — APPROVED** [review]
IIUC all the changes in this PR were already reviewed and approved in other PRs. So I'm good with merging this.

Not sure about the merge commit though. You may want to rebase this to follow the same PR process as gccrs

**Comment by CohenArthur** [comment]
@flip1995 thanks, I rebased and also added *one* commit that wasn't through a PR which helps when finding executable binaries: 7969bad

File: Cargo.toml [hunk 1]
```diff
@@ -16,3 +16,4 @@ getopts = "0.2"
 tempdir = "0.3"
 goblin = "0.4"
 ar = "0.8"
+is_executable = "1.0.1"
```
Comment (flip1995) [inline Cargo.toml:19]: And once again a crate that only exists, because Windows is weird :smile:

**Review by flip1995 — APPROVED** [review]
LGTM

**Pull request merged**